plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoClientModule.java

```diff
@@ -46,24 +50,33 @@ public static MongoSession createMongoSession(TypeManager typeManager, MongoClie
     {
         requireNonNull(config, "config is null");
 
-        MongoClientOptions.Builder options = MongoClientOptions.builder();
-
-        options.connectionsPerHost(config.getConnectionsPerHost())
-                .connectTimeout(config.getConnectionTimeout())
-                .socketTimeout(config.getSocketTimeout())
-                .socketKeepAlive(config.getSocketKeepAlive())
-                .sslEnabled(config.getSslEnabled())
-                .maxWaitTime(config.getMaxWaitTime())
-                .maxConnectionIdleTime(config.getMaxConnectionIdleTime())
-                .minConnectionsPerHost(config.getMinConnectionsPerHost())
+        MongoClientSettings.Builder options = MongoClientSettings.builder();
+        options.writeConcern(config.getWriteConcern().getWriteConcern())
                 .readPreference(config.getReadPreference().getReadPreference())
-                .writeConcern(config.getWriteConcern().getWriteConcern());
+                .applyToConnectionPoolSettings(builder -> builder.maxConnectionIdleTime(config.getMaxConnectionIdleTime(), SECONDS)
+                        .maxWaitTime(config.getMaxWaitTime(), SECONDS)
+                        .minSize(config.getMinConnectionsPerHost())
+                        .maxSize(config.getConnectionsPerHost()))
+                .applyToSocketSettings(builder -> builder.connectTimeout(config.getConnectionTimeout(), SECONDS))
+                .applyToSslSettings(builder -> builder.enabled(config.getSslEnabled()));
 
         if (config.getRequiredReplicaSetName() != null) {
-            options.requiredReplicaSetName(config.getRequiredReplicaSetName());
+            options.applyToClusterSettings(builder -> builder.requiredReplicaSetName(config.getRequiredReplicaSetName()));
+        }
+
+        if (!config.getConnectionUrl().isPresent() || !config.getConnectionUrl().get().isEmpty()) {
+            if (!config.getCredentials().isEmpty()) {
+                options.credential(config.getCredentials().get(0));
+            }
+
+            options.applyToClusterSettings(builder -> builder.mode(ClusterConnectionMode.SINGLE)
+                    .hosts(config.getSeeds()));
+        }
+        else {
+            options.applyConnectionString(new ConnectionString(config.getConnectionUrl().get()));
```

Comment:
This `if-else` block is wrong. Please fix like below:
```java
        if (config.getConnectionUrl().isPresent()) {
            options.applyConnectionString(new ConnectionString(config.getConnectionUrl().get()));
        }
        else {
            options.applyToClusterSettings(builder -> builder.mode(ClusterConnectionMode.SINGLE).hosts(config.getSeeds()));
        }
```

Reply:
the connection string could be present but empty

Reply:
Please see https://github.com/trinodb/trino/pull/9819#discussion_r752773086

Reply:
If  using seeds the credentials should be here if they are not empty